fedimint-client/src/lib.rs

```diff
@@ -1758,7 +1758,7 @@ impl ClientBuilder {
             module_inits: client.module_inits.clone(),
             primary_module_instance: Some(client.primary_module_instance),
             admin_creds: None,
-            db: client.db.clone(),
+            db_no_encoders: client.db.with_decoders(Default::default()),
```

Comment:
Notice - in `ClientBuilder`, the `db` should not contain any built-in decoders. It's somewhat important, and in here e.g. we would unnecessarily inherit decoders from previous `Client`, which is harmless right now, but could lead to bugs. For this reason I decided to rename it to this somewhat odd looking name.

Reply:
Now that the client can shut down cleanly we should get rid of `from_existing`, that was our old workaround to not have to re-open the DB.